packages/webapp/src/containers/Finances/RevenueDetail/index.jsx

```diff
@@ -0,0 +1,143 @@
+
+  const revenueTypes = useSelector(revenueTypeTileContentsSelector);
+  // Doesnt populate unless used with hook form persist (but consider adding back persist because while this form doesnt need to persist -- others might want to)
+  // const revenueTypes = useSortedRevenueTypes();
+  const sale = useSelector(revenueByIdSelector(sale_id));
+  const revenueType = useSelector(revenueTypeByIdSelector(sale?.revenue_type_id));
+
+  // Review after merging LF-3595
+  // Handles changing the type
+  const [selectedRevenueType, setSelectedRevenueType] = useState(revenueType);
+  useEffect(() => {
+    if (!sale) {
+      history.replace('/unknown_record');
+    }
+  }, [sale, history]);
+
+  // Dropdown should include the current revenue's type even if it has been retired
+  const revenueTypesArray = revenueTypes?.concat(revenueType?.deleted ? revenueType : []);
+
+  const revenueTypeReactSelectOptions = revenueTypesArray?.map((type) => {
+    const retireSuffix = type.deleted ? ` ${t('REVENUE.EDIT_REVENUE.RETIRED')}` : '';
+
+    return {
+      value: type.revenue_type_id,
+      label: type.farm_id
+        ? type.revenue_name + retireSuffix
+        : t(`revenue:${type.revenue_translation_key}`),
+    };
+  });
+
+  const onSubmit = (data) => {
+    let editedSale = {
+      sale_id: sale_id,
+      customer_name: data.customer_name,
+      sale_date: data.sale_date,
+      revenue_type_id: data.revenue_type_id.value,
+      note: data.note ? data.note : null,
+    };
+    const newRevenueType = revenueTypes.find(
+      (type) => type.revenue_type_id === data.revenue_type_id.value,
+    );
+    if (newRevenueType.crop_generated) {
+      editedSale.value = null;
+      editedSale.crop_variety_sale = Object.values(data.crop_variety_sale).map((c) => {
+        return {
+          sale_value: c.sale_value,
+          quantity: c.quantity,
+          quantity_unit: c.quantity_unit.label,
+          crop_variety_id: c.crop_variety_id,
+        };
+      });
+    } else if (!newRevenueType.crop_generated) {
+      editedSale.crop_variety_sale = null;
+      editedSale.value = data.value;
+    }
+
+    dispatch(updateSale(editedSale));
+  };
+
+  const handleEdit = () => {
+    dispatch(setPersistedPaths([`/revenue/${sale_id}/edit`]));
+    history.push(`/revenue/${sale_id}/edit`);
+  };
+
+  const onRetire = () => {
+    dispatch(deleteSale(sale_id));
+  };
+
+  const handleGoBack = () => {
+    history.back();
+  };
+
+  const onTypeChange = (typeId, setValue, REVENUE_TYPE_ID) => {
+    const newType = revenueTypeReactSelectOptions?.find((option) => option.value === typeId);
+    setValue(REVENUE_TYPE_ID, newType);
+    const revenueType = revenueTypes.find((type) => type.revenue_type_id === typeId);
+    setSelectedRevenueType(revenueType);
+  };
+
+  return (
+    <GeneralRevenue
+      key={isEditing ? 'editing' : 'readonly'}
+      onSubmit={isEditing ? onSubmit : undefined}
+      title={t('SALE.EDIT_SALE.TITLE')}
```

Comment:
Just a minor point after Sayaka and Anto did all the heavy lifting 😁... but why does the readonly view have the same title as the Edit view? Shouldn't this be "Sale Detail"?

<img width="451" alt="Screenshot 2023-10-20 at 11 02 56 AM" src="https://github.com/LiteFarmOrg/LiteFarm/assets/82857964/febaf05c-3f23-4019-a3de-302296d51921">

Reply:
Yeah I can change that